src/jetstream/plugins/epinio/dex/api.go

```diff
@@ -0,0 +1,43 @@
+package epiniodex
+
+import (
+	"net/http"
+
+	"github.com/epinio/ui-backend/src/jetstream/plugins/epinio/rancherproxy/api"
+
+	jInterfaces "github.com/epinio/ui-backend/src/jetstream/repository/interfaces"
+
+	"github.com/labstack/echo/v4"
+)
+
+type RedirectUrlResponse struct {
+	RedirectUrl string `json:"redirectUrl"`
+}
+
+func RedirectUrl(ec echo.Context, p jInterfaces.PortalProxy) error {
+	oidcProvider, err := p.GetDex()
+
+	if err != nil {
+		return jInterfaces.NewHTTPShadowError(
+			http.StatusInternalServerError,
+			"Failed to create Dex Client",
+			"Failed to create Dex Client: %+v",
+			err,
+		)
+	}
+
+	state := ec.QueryParams().Get("state")
+	if len(state) == 0 {
+		return jInterfaces.NewHTTPShadowError(
+			http.StatusInternalServerError,
+			"Invalid request, `state` required",
+			"Invalid request, `state` required",
+		)
+	}
+
+	dexUrl, _ := oidcProvider.AuthCodeURLWithPKCE(state)
```

Comment:
I have the feeling that you should not use the PKCE flow, since you have a secret and it's a server-to-server request. You can probably use directly the `oidcProvider.Config.AuthCodeURL(state)`

https://dexidp.io/docs/using-dex/

The PKCE flow is needed when you cannot embed a secret safely (in our cli for example, or in a mobile application). If it works you can drop a lot of unneeded files.

Reply:
Ah, yeah. The secret wasn't that safe before but now it is.

Is there any inherent downside to leaving this in for the moment?

Reply:
The PKCE is a bit less secure I guess. It's not because of it, but just because usually temporary means forever.

Do you think is possible to do a quick change and test? If it's complicated we can try to revisit in a second quick round. We could maybe do this by ourself, and your review.

Reply:
Hum, having a look [here](https://www.scottbrady91.com/oauth/client-authentication-vs-pkce) maybe it's even better to have both. So let's keep it like this!